src/sys/socket/mod.rs

```diff
@@ -405,17 +423,17 @@ impl<'a> ControlMessage<'a> {
         }
     }
 
-    // Unsafe: start and end of buffer must be size_t-aligned (that is,
-    // cmsg_align'd). Updates the provided slice; panics if the buffer
-    // is too small.
+    // Unsafe: start and end of buffer must be cmsg_align'd. Updates
+    // the provided slice; panics if the buffer is too small.
     unsafe fn encode_into<'b>(&self, buf: &mut &'b mut [u8]) {
         match *self {
             ControlMessage::ScmRights(fds) => {
-                let cmsg = cmsghdr {
-                    cmsg_len: self.len() as type_of_cmsg_len,
-                    cmsg_level: libc::SOL_SOCKET,
-                    cmsg_type: libc::SCM_RIGHTS,
-                    cmsg_data: [],
+                let cmsg = {
+                  let mut cmsg: cmsghdr = mem::uninitialized();
```

Comment:
This would be more clear if rewritten to use struct update syntax:

```rust
    let mut cmsg = cmsghdr {
        cmsg_len: self.len() as _,
        cmsg_level: libc::SOL_SOCKET,
        cmsg_type: libc::SCM_RIGHTS,
        ..unsafe { mem::uninitialized() }
    };
```

Reply:
I agree that's nicer, but it looks like the musl target prevents this due to private fields. 😞

Reply:
It's only a problem for `msghdr`, so I'll keep the `cmsghdr` changes and revert the rest.